addons/mrp/views/mrp_production_views.xml

```diff
@@ -513,7 +513,7 @@
                     <field name="name"/>
                     <field name="product_id"/>
                     <field name="product_qty"/>
-                    <field name="product_uom_id" options="{'no_open':True,'no_create':True}"/>
+                    <field name="product_uom_id" options="{'no_open': True, 'no_create': True}" groups="uom.group_uom" readonly="state != 'draft'" class="flex-grow-0" style="width:auto!important"/>
```

Comment:
In a kanban view, fields outside of the template aren't rendered, so this change doesn't make sense.
Also, because of the `groups`, you aren't even able to display the MO's kanban view if UoM setting is not active...